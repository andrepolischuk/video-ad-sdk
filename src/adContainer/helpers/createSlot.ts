export const createSlot = (
  placeholder: HTMLElement,
  width: number,
  height: number
): HTMLElement => {
  const slot = document.createElement('DIV')

  Object.assign(slot.style, {
    border: '0px',
    cursor: 'pointer',
    height: `${height}px`,
    left: '0px',
    margin: '0px',
    padding: '0px',
    position: 'absolute',
    top: '0px',
    width: `${width}px`
  })

  placeholder.appendChild(slot)

  return slot
}
