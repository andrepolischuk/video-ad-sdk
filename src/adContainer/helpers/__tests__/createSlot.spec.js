import createSlot from '../createSlot';

describe('createSlot', () => {
  test('must return a slot element', () => {
    const slot = createSlot(document.body, 300, 200);

    expect(slot).toBeInstanceOf(HTMLDivElement);
    expect(slot).toMatchSnapshot();

    slot.parentElement.removeChild(slot);
  });
});
