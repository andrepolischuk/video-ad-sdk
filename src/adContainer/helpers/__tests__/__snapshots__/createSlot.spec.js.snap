// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSlot must return a slot element 1`] = `
<div
  style="border: 0px; cursor: pointer; height: 200px; left: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; width: 300px;"
/>
`;
